Disable the hardware ECC checking on S3C2440 based platforms (HXD8, SMDK2440,
GTA02) for the time being, since our u-boot doesn't yet support it for 2k page
size NAND

Index: linux-2.6.21-moko/drivers/mtd/nand/s3c2410.c
===================================================================
--- linux-2.6.21-moko.orig/drivers/mtd/nand/s3c2410.c
+++ linux-2.6.21-moko/drivers/mtd/nand/s3c2410.c
@@ -644,7 +644,7 @@
 	nmtd->mtd.owner    = THIS_MODULE;
 	nmtd->set	   = set;
 
-	if (hardware_ecc) {
+	if (info->cpu_type == TYPE_S3C2410 && hardware_ecc) {
 		chip->ecc.calculate = s3c2410_nand_calculate_ecc;
 		chip->ecc.correct   = s3c2410_nand_correct_data;
 		chip->ecc.mode	    = NAND_ECC_HW;
